Framework/API/inc/MantidAPI/Workspace.h

```diff
@@ -117,7 +117,7 @@ class MANTID_API_DLL Workspace : public Kernel::DataItem {
   /// The history of the workspace, algorithm and environment
   std::unique_ptr<WorkspaceHistory> m_history;
   /// Storage mode of the Workspace (used for MPI runs)
-  Parallel::StorageMode m_storageMode{Parallel::StorageMode::Cloned};
+  Parallel::StorageMode m_storageMode{Parallel::StorageMode::MasterOnly};
```

Comment:
I changed this such that workspaces created in `execNonMaster` get the right storage mode by default. For properly initialized workspaces (unlike those on non-master ranks if execution mode is `MasterOnly`), the storage mode is set via `IndexInfo` that is passed to `create<MatrixWorkspace>`.

Open problem/question:
- How should the storage mode of non-`MatrixWorkspaces` be set? I do not want to provide a public `setStorageMode`, since it cannot change after construction. Probably using a factory function or method?